### ISSUE TITLE
[Sources] Remove contents.get calls

### ISSUE THREAD

**Issue by jasonLaster** [devtools-html/debugger.html#5738]
We recently refactored our immutable Source object to be a Record instead of a Map. As a result, we can now replace `contents.get("id")` calls with `contents.id` calls. 

This refactor also removes some oversights in the code too :)

```diff
diff --git a/src/components/PrimaryPanes/SourcesTree.js b/src/components/PrimaryPanes/SourcesTree.js
index 078096d..2a90af5 100644
--- a/src/components/PrimaryPanes/SourcesTree.js
+++ b/src/components/PrimaryPanes/SourcesTree.js
@@ -163,28 +163,20 @@ class SourcesTree extends Component<Props, State> {
 
   selectItem = item => {
     if (!nodeHasChildren(item)) {
-      this.props.selectLocation({ sourceId: item.contents.get("id") });
+      this.props.selectLocation({ sourceId: item.contents.id });
     }
   };
 
   getPath = item => {
-    const { sources } = this.props;
-    const obj = item.contents.get && item.contents.get("id");
-
-    let blackBoxedPart = "";
-
-    if (
-      typeof obj !== "undefined" &&
-      sources.has(obj) &&
-      sources.get(obj).get("isBlackBoxed")
-    ) {
-      blackBoxedPart = "update";
+    if (nodeHasChildren(item)) {
+      return item.path;
     }
 
+    const blackBoxedPart = item.contents.isBlackBoxed ? "update" : "";
     return `${item.path}/${item.name}/${blackBoxedPart}`;
   };
 
-  getIcon = (sources, item, depth) => {
+  getIcon = (item, depth) => {
     const { debuggeeUrl, projectRoot } = this.props;
 
     if (item.path === "/Webpack") {
@@ -205,9 +197,8 @@ class SourcesTree extends Component<Props, State> {
     }
 
     if (!nodeHasChildren(item)) {
-      const obj = item.contents.get("id");
-      const source = sources.get(obj);
-      if (source && source.get("isBlackBoxed")) {
+      const source = item.contents;
+      if (source && source.isBlackBoxed) {
         return <img className="blackBox" />;
       }
       const sourceType = {
@@ -239,7 +230,7 @@ class SourcesTree extends Component<Props, State> {
     const menuOptions = [];
 
     if (!isDirectory(item)) {
-      const source = item.contents.get("url");
+      const source = item.contents.url;
       const copySourceUri2 = {
         id: "node-menu-copy-source",
         label: copySourceUri2Label,
@@ -287,8 +278,7 @@ class SourcesTree extends Component<Props, State> {
       <i className="no-arrow" />
     );
 
-    const { sources } = this.props;
-    const icon = this.getIcon(sources, item, depth);
+    const icon = this.getIcon(item, depth);
 
     return (
```

**Comment by jasonLaster** [comment]
@lukaszsobek un-assigning as it's been 2 weeks.

**Comment by Hindelburg** [comment]
I just want to confirm that the bug only occurs in file /src/components/PrimaryPanes/SourcesTree.js. If it is found somewhere else as well, could you please give an idea on where to look. 

Thanks in advance, Hindelburg.

**Comment by jasonLaster** [comment]
I just do cmd+shift+f and search `.get`

**Comment by Hindelburg** [comment]
We have made the changes to /src/components/PrimaryPanes/SourcesTree.js so that it used contents.id and we searched through the rest of the repository and didn't find anything that needed changes. We also ran the tests and they passed. Is there anything else we should do before submitting a pull request?

**Comment by jasonLaster** [comment]
submit it!

**Comment by jasonLaster** [comment]
Here is a good patch for removing .get
https://gist.github.com/f8efdd61c9708e34b871770bfcba4647

**Comment by jasonLaster** [comment]
This will be fixed here #6326

**Comment by jasonLaster** [comment]
@Hindelburg i recently added some new help wanted bugs
